### PR TITLE
Fix analyzer warnings

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
lgtm

File: sky/sdk/example/raw/mutating-dom.dart [hunk 1]
```diff
@@ -148,26 +148,27 @@ void doFrame(double timeStamp) {
         break;
       }
     } else {
-      assert(node is sky.Text); // 
+      assert(node is sky.Text); //
```
Comment (Hixie) [inline sky/sdk/example/raw/mutating-dom.dart]: remove trailing //

File: sky/sdk/example/raw/mutating-dom.dart [hunk 2]
```diff
@@ -148,26 +148,27 @@ void doFrame(double timeStamp) {
...
+      final sky.Text text = node;
```
Comment (Hixie) [inline sky/sdk/example/raw/mutating-dom.dart]: and a `// TODO(ianh): remove this once analyzer is cleverer`

File: sky/sdk/example/raw/mutating-dom.dart [hunk 3]
```diff
@@ -178,7 +179,7 @@ void doFrame(double timeStamp) {
   int count = 1;
   while (node != null) {
     if (node is sky.Element && node.firstChild != null) {
-      node = node.firstChild;
+      node = (node as sky.Element).firstChild;
```
Comment (Hixie) [inline sky/sdk/example/raw/mutating-dom.dart]: add same TODO(ianh) here

**Pull request merged**